lib/reply.js

```diff
@@ -157,19 +157,20 @@ Reply.prototype.send = function (payload) {
       payload = this[kReplySerializer](payload)
     }
 
-  // The indexOf below also matches custom json mimetypes such as 'application/hal+json' or 'application/ld+json'
-  } else if (hasContentType === false || contentType.indexOf('json') > -1) {
-    if (hasContentType === false) {
-      this[kReplyHeaders]['content-type'] = CONTENT_TYPE.JSON
+  // The 'includes' below also matches custom json mimetypes such as 'application/hal+json' or 'application/ld+json'
+  } else if (!hasContentType || contentType.includes('json')) {
+    if (!hasContentType) {
+      this.type(CONTENT_TYPE.JSON)
     } else {
       // If user doesn't set charset, we will set charset to utf-8
-      if (contentType.indexOf('charset') === -1) {
+      if (!contentType.includes('charset')) {
```

Comment:
`includes` is _really slow_, use `indexOf`

Reply:
[In browser tests](https://perf.link/#eyJpZCI6ImsxbmJoNDlxOXFxIiwidGl0bGUiOiJpbmRleE9mIC8gaW5jbHVkZXMiLCJiZWZvcmUiOiJmdW5jdGlvbiBnZXRSYW5kb21TdHIoKSB7XG4gICAgcmV0dXJuIE1hdGgucmFuZG9tKCkudG9TdHJpbmcoMTYpLnN1YnN0cmluZygyKVxufVxuXG5mdW5jdGlvbiBnZXRSYW5kb21JbnQobWluLCBtYXgpIHtcbiAgICByZXR1cm4gbWluICsgTWF0aC5mbG9vcihNYXRoLnJhbmRvbSgpICogKG1heCAtIG1pbikpO1xufVxuXG5jb25zdCB3b3JkcyA9IFtdXG5cbmZvciAobGV0IGkgPSAwOyBpIDwgMTAwMDsgaSsrKSB7XG4gICAgd29yZHMucHVzaChnZXRSYW5kb21TdHIoKSlcbn1cblxuY29uc3Qgd29yZHNTdHIgPSB3b3Jkcy5qb2luKCcnKTsiLCJ0ZXN0cyI6W3sibmFtZSI6IlN0cmluZy5wcm90b3R5cGUuaW5kZXhPZigpIiwiY29kZSI6ImNvbnN0IGxvb2t1cFdvcmQgPSB3b3Jkc1tnZXRSYW5kb21JbnQoMCwgd29yZHMubGVuZ3RoKV1cbnJldHVybiB3b3Jkc1N0ci5pbmRleE9mKGxvb2t1cFdvcmQpICE9PSAtMSIsInJ1bnMiOlsxMDAwLDEwMDAsMTAwMCwxMDAwLDEwMDAsMTAwMCwxMDAwLDEwMDAsMTAwMCwxMDAwLDEwMDAsMTAwMCwxMDAwLDEwMDAsMTAwMCwxMDAwLDEwMDAsMTAwMCwxMDAwLDEwMDAsMTAwMCwxMDAwLDEwMDAsMTAwMCwxMDAwLDEwMDAsMTAwMCwxMDAwLDEwMDAsMTAwMCwxMDAwLDEwMDAsMTAwMCwxMDAwLDEwMDAsMTAwMCwxMDAwLDEwMDAsMTAwMCwxMDAwLDEwMDAsMTAwMCwxMDAwLDEwMDAsMTAwMCwxMDAwLDEwMDAsMTAwMCwxMDAwLDEwMDAsMTAwMCwxMDAwLDEwMDAsMTAwMCwxMDAwLDEwMDAsMTAwMCwxMDAwLDEwMDAsMTAwMCwxMDAwLDEwMDAsMTAwMCwxMDAwLDEwMDAsMTAwMCwxMDAwLDEwMDAsMTAwMCwxMDAwLDEwMDAsMTAwMCwxMDAwLDEwMDAsMTAwMCwxMDAwLDEwMDAsMTAwMCwxMDAwLDEwMDAsMTAwMCwxMDAwLDEwMDAsMTAwMCwxMDAwLDEwMDAsMTAwMCwxMDAwLDEwMDAsMTAwMCwxMDAwLDEwMDAsMTAwMCwxMDAwLDEwMDAsMTAwMCwxMDAwLDEwMDAsMTAwMCwxMDAwXSwib3BzIjoxMDAwfSx7Im5hbWUiOiJTdHJpbmcucHJvdG90eXBlLmluY2x1ZGVzKCkiLCJjb2RlIjoiY29uc3QgbG9va3VwV29yZCA9IHdvcmRzW2dldFJhbmRvbUludCgwLCB3b3Jkcy5sZW5ndGgpXVxucmV0dXJuIHdvcmRzU3RyLmluY2x1ZGVzKGxvb2t1cFdvcmQpOyIsInJ1bnMiOlsxMDAwLDEwMDAsMTAwMCwxMDAwLDEwMDAsMTAwMCwxMDAwLDEwMDAsMTAwMCwxMDAwLDEwMDAsMTAwMCwxMDAwLDEwMDAsMTAwMCwxMDAwLDEwMDAsMTAwMCwxMDAwLDEwMDAsMTAwMCwxMDAwLDEwMDAsMTAwMCwxMDAwLDEwMDAsMTAwMCwxMDAwLDEwMDAsMTAwMCwxMDAwLDEwMDAsMTAwMCwxMDAwLDEwMDAsMTAwMCwxMDAwLDEwMDAsMTAwMCwxMDAwLDEwMDAsMTAwMCwxMDAwLDEwMDAsMTAwMCwxMDAwLDEwMDAsMTAwMCwxMDAwLDEwMDAsMTAwMCwxMDAwLDEwMDAsMTAwMCwxMDAwLDEwMDAsMTAwMCwxMDAwLDEwMDAsMTAwMCwxMDAwLDEwMDAsMTAwMCwxMDAwLDEwMDAsMTAwMCwxMDAwLDEwMDAsMTAwMCwxMDAwLDEwMDAsMTAwMCwxMDAwLDEwMDAsMTAwMCwxMDAwLDEwMDAsMTAwMCwxMDAwLDEwMDAsMTAwMCwxMDAwLDEwMDAsMTAwMCwxMDAwLDEwMDAsMTAwMCwxMDAwLDEwMDAsMTAwMCwxMDAwLDEwMDAsMTAwMCwxMDAwLDEwMDAsMTAwMCwxMDAwLDEwMDAsMTAwMCwxMDAwXSwib3BzIjoxMDAwfV0sInVwZGF0ZWQiOiIyMDIyLTA2LTIyVDEzOjE4OjMxLjA0OFoifQ%3D%3D), I also checked in node 16, results are similar to browser.

Reply:
[In browser tests](https://perf.link/#eyJpZCI6ImsxbmJoNDlxOXFxIiwidGl0bGUiOiJpbmRleE9mIC8gaW5jbHVkZXMiLCJiZWZvcmUiOiJmdW5jdGlvbiBnZXRSYW5kb21TdHIoKSB7XG4gICAgcmV0dXJuIE1hdGgucmFuZG9tKCkudG9TdHJpbmcoMTYpLnN1YnN0cmluZygyKVxufVxuXG5mdW5jdGlvbiBnZXRSYW5kb21JbnQobWluLCBtYXgpIHtcbiAgICByZXR1cm4gbWluICsgTWF0aC5mbG9vcihNYXRoLnJhbmRvbSgpICogKG1heCAtIG1pbikpO1xufVxuXG5jb25zdCB3b3JkcyA9IFtdXG5cbmZvciAobGV0IGkgPSAwOyBpIDwgMTAwMDsgaSsrKSB7XG4gICAgd29yZHMucHVzaChnZXRSYW5kb21TdHIoKSlcbn1cblxuY29uc3Qgd29yZHNTdHIgPSB3b3Jkcy5qb2luKCcnKTsiLCJ0ZXN0cyI6W3sibmFtZSI6IlN0cmluZy5wcm90b3R5cGUuaW5kZXhPZigpIiwiY29kZSI6ImNvbnN0IGxvb2t1cFdvcmQgPSB3b3Jkc1tnZXRSYW5kb21JbnQoMCwgd29yZHMubGVuZ3RoKV1cbndvcmRzU3RyLmluZGV4T2YobG9va3VwV29yZCkgIT09IC0xIiwicnVucyI6WzcwMDAsMjQ3MDAwLDU0MDAwLDY3MDAwLDkwMDAsMjQxMDAwLDE0NDAwMCwxNDAwMDAsNTAwMCw2MDAwLDE4NDAwMCwyMDAwMDAsMzYwMDAsMjIxMDAwLDUzMDAwLDE0NTAwMCwyNTcwMDAsMjQwMDAwLDEzNDAwMCwxNDAwMDAsNzEwMDAsMTczMDAwLDE4MzAwMCwyMzMwMDAsMTM3MDAwLDc0MDAwLDE4NjAwMCwxMTEwMDAsNDIwMDAsMzIwMDAsMTE2MDAwLDE1NDAwMCwyNjUwMDAsOTAwMCwxMjIwMDAsMTA2MDAwLDEwMDAsMjI4MDAwLDI1ODAwMCwyMjkwMDAsMTE4MDAwLDIxODAwMCwyMjAwMDAsNjkwMDAsNDEwMDAsNTkwMDAsMzEwMDAsMjYwMDAwLDE3MTAwMCw3MDAwLDI0OTAwMCw5MTAwMCwxNjEwMDAsMTU5MDAwLDYyMDAwLDIwMzAwMCwxNDIwMDAsMjYzMDAwLDE0MDAwLDEyODAwMCwxMjEwMDAsMTIzMDAwLDIxOTAwMCwyMDAwMCwyNDkwMDAsNTQwMDAsMjM4MDAwLDUwMDAsNjEwMDAsMzcwMDAsMTc1MDAwLDIxMTAwMCwzMjAwMCw4ODAwMCwxMDAwLDExMjAwMCwxNTAwMDAsMjAzMDAwLDg5MDAwLDE1NzAwMCwxNDUwMDAsODcwMDAsNDgwMDAsMzAwMCw2MzAwMCwyMzAwMDAsMjM5MDAwLDE1MzAwMCwxODMwMDAsNjIwMDAsMTE0MDAwLDI2MzAwMCw0MDAwMCwxNjMwMDAsMTkxMDAwLDIzMDAwLDY2MDAwLDE1MjAwMCwxMDAwMCwyOTAwMF0sIm9wcyI6MTI1MzUwfSx7Im5hbWUiOiJTdHJpbmcucHJvdG90eXBlLmluY2x1ZGVzKCkiLCJjb2RlIjoiY29uc3QgbG9va3VwV29yZCA9IHdvcmRzW2dldFJhbmRvbUludCgwLCB3b3Jkcy5sZW5ndGgpXVxud29yZHNTdHIuaW5jbHVkZXMobG9va3VwV29yZCkiLCJydW5zIjpbOTAwMCwyNDAwMDAsNDkwMDAsMTk1MDAwLDcwMDAsMTY3MDAwLDI0OTAwMCwxOTAwMCw0MDAwLDE4NzAwMCwyMTcwMDAsMTM2MDAwLDYyMDAwLDEyNzAwMCwxNjMwMDAsMjYwMDAsMjIyMDAwLDExMDAwLDIzNzAwMCwxMjkwMDAsMzAwMCwzNjAwMCwxOTkwMDAsMTE3MDAwLDE5MjAwMCwyMDMwMDAsMzkwMDAsMjMwMDAwLDE3OTAwMCwxODUwMDAsMTA3MDAwLDU5MDAwLDIxNDAwMCwzOTAwMCwxMzUwMDAsNjIwMDAsMjIyMDAwLDE1OTAwMCwyMzcwMDAsMTEyMDAwLDI3MTAwMCwxMDEwMDAsNzAwMCwyMDAwLDQ2MDAwLDE1OTAwMCwxMDcwMDAsODUwMDAsMTUzMDAwLDE3NDAwMCwxMTQwMDAsNDQwMDAsMTE4MDAwLDU5MDAwLDE4NjAwMCwxMTAwMDAsOTcwMDAsMTIyMDAwLDE2NTAwMCwyMTcwMDAsOTYwMDAsMjE5MDAwLDEzODAwMCw2NzAwMCwxMTMwMDAsMTY2MDAwLDE0OTAwMCw4NjAwMCwxNTkwMDAsNDkwMDAsMTY0MDAwLDMyMDAwLDIxNDAwMCw3MTAwMCwyNDYwMDAsMTAwMDAwLDE1MjAwMCwxNzEwMDAsODYwMDAsNDUwMDAsMTY4MDAwLDk3MDAwLDE0OTAwMCwxNTcwMDAsMTgyMDAwLDIwMjAwMCwxMTAwMCwzMTAwMCwzMTAwMCwyMjIwMDAsMTU1MDAwLDExMzAwMCw0MjAwMCwxMDAwLDEwMzAwMCwxMDEwMDAsMTIwMDAsMTQ0MDAwLDUwMDAsMjI2MDAwXSwib3BzIjoxMjE5NzB9XSwidXBkYXRlZCI6IjIwMjItMDYtMjJUMTM6MjQ6MzAuMDgyWiJ9) I also checked it in node 16, results are similar to browser

Reply:
Your test include the performance of word selection.
The word selection should be moved to global.

Depends on where the `word` position. `.includes` is generally slower than `.indexOf`.
`100%` vs `90 - 73%` and `90%` vs `100%`

Reply:
Yeah, I intentionally put it inside the function, just in case (to prevent browsers optimizations).
> Depends on where the word position. .includes is generally slower than .indexOf.
100% vs 90 - 73% and 90% vs 100%

ok, I'll revert it then :ok_hand: